samples/apps/smallbank/app/smallbank.cpp

```diff
@@ -486,7 +486,7 @@ namespace ccfapp
   };
 
   std::shared_ptr<ccf::UserRpcFrontend> get_rpc_handler(
-    NetworkTables& nwt, AbstractNotifier& notifier)
+    NetworkTables& nwt, ccfapp::AbstractNodeContext& context)
```

Comment:
Making the historical state cache available is a breaking change for all apps. Rather than adding an argument every time we add a system like this (changing the signature all apps need to implement), I've wrapped these in a single `context`. New services are exposed as new functions on this context, but apps that don't use them can upgrade with no changes.